backend/inspirehep/records/views.py

```diff
@@ -239,6 +246,32 @@ def reference_self_curation(args):
     return jsonify({"message": "Success"}), 200
 
 
+@blueprint.route("/literature/<path:recid>/<int:old_revision>..<int:new_revision>")
+@login_required_with_roles([Roles.superuser.value, Roles.cataloger.value])
+def reference_diff(recid, old_revision, new_revision):
+    if new_revision <= old_revision:
+        return (
+            jsonify({"message": "Old revision must be lower than new revision!"}),
+            400,
+        )
+    new_version_id = new_revision + 1
+    old_version_id = old_revision + 1
+    record_uuid = str(PidStoreBase.get_uuid_for_recid(recid, "lit"))
+    try:
+        old_record = InspireRecord.get_record(
+            record_uuid, record_version=old_version_id
+        )
+        new_record = InspireRecord.get_record(
+            record_uuid, record_version=new_version_id
+        )
+    except StaleDataError:
```

Comment:
this wouldn't return `NoResultFound` if the version doesn't exist?

Reply:
no, we convert `NoResultFound` to `StaleDataError` in records api 🙃